bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockBookKeeper.java

```diff
@@ -49,7 +48,10 @@
  */
 public class MockBookKeeper extends BookKeeper {
 
-    final ExecutorService executor = Executors.newFixedThreadPool(1, new DefaultThreadFactory("mock-bookkeeper"));
+    final OrderedExecutor orderedExecutor = OrderedExecutor.newBuilder()
```

Comment:
Why change this thread pool?

Reply:
We need call `getMainWorkerPool()`  in `LedgerHandle` constructor method. If we didn't change it, the `getMainWorkerPool()` will return null and throw NPE in the constructor method:

[Line199](https://github.com/apache/bookkeeper/pull/3738/files#diff-1d893bb31553b5e1f55c8301d04ae15f38e0d35f531f9dd22475128b7972ddf9R197-R199)